packages/hash/api/src/graph/workspace-types.ts

```diff
@@ -314,6 +330,35 @@ const blockEntityTypeInitializer = async (graphApi: GraphApi) => {
   })(graphApi);
 };
 
+const tokensPropertyTypeInitializer = propertyTypeInitializer({
+  namespace: WORKSPACE_ACCOUNT_SHORTNAME,
+  title: "Text Tokens",
+  /** @todo: properly type this */
```

Comment:
I thought about this since, and I'm now wondering whether we want to type this as anything more than an array of objects? Currently the definition looks [like this](https://github.com/hashintel/hash/blob/5dbf065354c1a031e2e39d6980a116281578e99f/packages/hash/shared/src/graphql/types.ts#L8-L18):
```TypeScript
type TextToken =
  | {
      tokenType: "text";
      text: string;
      bold?: boolean;
      italics?: boolean;
      underline?: boolean;
      link?: string;
    }
  | { tokenType: "hardBreak" }
  | { tokenType: "mention"; mentionType: "user"; entityId: string };
```

If we were to define each of these with corresponding property types, the workspace FE will lose a lot of flexibility with what is stored in a text token. Adding a property would require defining a whole new property type, accessing a property would require the base URI of the property type (i.e. `properties["http://.../token"][0]["http://.../text/"]`).

If we do want the final text tokens to be defined of nested property types, happy to create the task and link it in the todo. I think it shouldn't be done pre-merge with `main` though as there would be a very large diff with the existing prosemirror/text-editing code.